### ISSUE TITLE
Is 0000001.0000002.00300003 a valid semver or not?

### ISSUE THREAD

**Issue by isaacs** [mojombo/semver#112]
Came up in the discussions in #110.

Are leading zeroes allowed or not?

I think that they shouldn't be allowed, because it's strange to me that `0000001.0000002.00300003` would be functionally identical to `1.2.300003`.

Same for numeric prerelease identifiers.  Is `1.2.3-beta.000000000000009` the same as `1.2.3-beta.9` and `1.2.3-beta.09`?

If leading zeros ARE allowed, then it should be made explicit.  I'd assumed that they weren't allowed, and I doubt whether most SemVer parsers do allow them, at least in the main version.

In other words, I'm suggesting that the regular expression for major/minor/patch version segments would be something like `/^[1-9]\d*|0$/`.
